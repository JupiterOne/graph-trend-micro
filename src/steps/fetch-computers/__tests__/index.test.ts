import {
  Recording,
  setupRecording,
  createMockStepExecutionContext,
} from '../../../../test';

import step, { createComputerEntity } from '../index';
import {
  createDeepSecurityClient,
  DeepSecurityComputer,
} from '../../../provider';
import { TrendMicroIntegrationConfig } from '../../../types';
import { omit } from 'lodash';

let recording: Recording;

beforeEach(() => {
  recording = setupRecording({
    directory: __dirname,
    name: 'computers',
  });
});

afterEach(async () => {
  await recording.stop();
});

test('computer fetching', async () => {
  const context = createMockStepExecutionContext<TrendMicroIntegrationConfig>({
    instanceConfig: {
      apiKey: 'apiKey',
    },
  });
  const provider = createDeepSecurityClient(context.instance);

  const results = await provider.listComputers();
  expect(
    results.computers
      .map(createComputerEntity)
      .map((e) => omit(e, ['_rawData'])),
  ).toMatchSnapshot();
});

test('computer entity conversion', () => {
  const computer = {
    hostName: 'ec2-54-187-35-33.us-west-2.compute.amazonaws.com',
    description:
      "This computer is a demonstration of Deep Security's capabilities. Open a browser to http://ec2-54-187-35-33.us-west-2.compute.amazonaws.com for more information.",
    policyID: 3,
    agentVersion: '11.1.0.163',
    platform: 'Amazon Linux (64 bit) (4.14.62-65.117.amzn1.x86_64)',
    relayListID: 0,
    agentFingerPrint:
      'F9:51:AE:21:1F:7E:08:91:D2:5A:3D:4D:0D:17:F3:53:E9:EA:4E:28',
    biosUUID: 'ec29a088-a562-cdad-48d9-d76b0e3385c4',
    hostGUID: '6BE38408-DA9E-4A03-1EB4-5CE824AD0C58',
    groupID: 25,
    ID: 1,
  } as DeepSecurityComputer;

  expect(createComputerEntity(computer)).toEqual({
    _key: 'trend-micro-computer:1',
    _type: 'trend_micro_computer',
    _class: ['Host', 'Device'],
    name: 'ec2-54-187-35-33.us-west-2.compute.amazonaws.com',
    displayName: 'ec2-54-187-35-33.us-west-2.compute.amazonaws.com',
    hostname: 'ec2-54-187-35-33.us-west-2.compute.amazonaws.com',
    platform: 'linux',
    id: 'trend-micro-computer:1',
    groupId: 'trend-micro-computer-group:25',
    description:
      "This computer is a demonstration of Deep Security's capabilities. Open a browser to http://ec2-54-187-35-33.us-west-2.compute.amazonaws.com for more information.",
    agentGUID: undefined,
    agentStatus: undefined,
    applianceStatus: undefined,
    awsAccountId: undefined,
    cloudProvider: undefined,
    createdOn: undefined,
    ec2InstanceId: undefined,
    hostGUID: '6BE38408-DA9E-4A03-1EB4-5CE824AD0C58',
    category: 'endpoint',
    make: null,
    model: null,
    serial: null,
    deviceId: '6BE38408-DA9E-4A03-1EB4-5CE824AD0C58',
    _rawData: [
      {
        name: 'default',
        rawData: computer,
      },
    ],
  });
});

test('step data collection', async () => {
  const context = createMockStepExecutionContext<TrendMicroIntegrationConfig>({
    instanceConfig: {
      apiKey: 'apiKey',
    },
  });
  await step.executionHandler(context);

  expect(context.jobState.collectedEntities).toHaveLength(3);
  expect(context.jobState.collectedRelationships).toHaveLength(0);

  expect(context.jobState.collectedEntities).toEqual([
    expect.objectContaining({
      _key: 'trend-micro-computer:1',
    }),
    expect.objectContaining({
      _key: 'trend-micro-computer:34',
    }),
    expect.objectContaining({
      _key: 'trend-micro-computer:35',
    }),
  ]);
});
