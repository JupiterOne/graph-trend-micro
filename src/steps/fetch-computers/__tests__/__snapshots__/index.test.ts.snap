// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computer fetching 1`] = `
Array [
  Object {
    "_class": Array [
      "Host",
      "Device",
    ],
    "_key": "trend-micro-computer:1",
    "_type": "trend_micro_computer",
    "agentGUID": undefined,
    "agentStatus": "active",
    "antiMalwareStatus": "active",
    "applianceStatus": undefined,
    "applicationControlStatus": "inactive",
    "awsAccountId": undefined,
    "category": "endpoint",
    "cloudProvider": undefined,
    "createdOn": undefined,
    "description": "This computer is a demonstration of Deep Security's capabilities. Open a browser to http://ec2-54-187-35-33.us-west-2.compute.amazonaws.com for more information.",
    "deviceId": "6BE38408-DA9E-4A03-1EB4-5CE824AD0C58",
    "displayName": "ec2-54-187-35-33.us-west-2.compute.amazonaws.com",
    "ec2InstanceId": undefined,
    "firewallStatus": "active",
    "groupId": undefined,
    "hostGUID": "6BE38408-DA9E-4A03-1EB4-5CE824AD0C58",
    "hostname": "ec2-54-187-35-33.us-west-2.compute.amazonaws.com",
    "id": "trend-micro-computer:1",
    "integrityMonitoringStatus": "active",
    "intrusionPreventionStatus": "active",
    "lastSeenOn": 1587565628816,
    "logInspectionStatus": "active",
    "make": null,
    "model": null,
    "name": "ec2-54-187-35-33.us-west-2.compute.amazonaws.com",
    "platform": "linux",
    "securityUpdateStatus": "warning",
    "serial": null,
    "webReputationStatus": "active",
  },
  Object {
    "_class": Array [
      "Host",
      "Device",
    ],
    "_key": "trend-micro-computer:34",
    "_type": "trend_micro_computer",
    "agentGUID": undefined,
    "agentStatus": "inactive",
    "antiMalwareStatus": "inactive",
    "applianceStatus": undefined,
    "applicationControlStatus": "inactive",
    "awsAccountId": "697977203433",
    "category": "endpoint",
    "cloudProvider": "AWS",
    "createdOn": undefined,
    "description": "",
    "deviceId": "FBA83AE4-7389-6448-F6E5-12E326BEBDBA",
    "displayName": "i-075776bd5b0b322aa",
    "ec2InstanceId": "i-075776bd5b0b322aa",
    "firewallStatus": "inactive",
    "groupId": "trend-micro-computer-group:79",
    "hostGUID": "FBA83AE4-7389-6448-F6E5-12E326BEBDBA",
    "hostname": "172.31.121.120",
    "id": "trend-micro-computer:34",
    "integrityMonitoringStatus": "inactive",
    "intrusionPreventionStatus": "inactive",
    "lastSeenOn": undefined,
    "logInspectionStatus": "inactive",
    "make": null,
    "model": null,
    "name": "i-075776bd5b0b322aa",
    "platform": "linux",
    "securityUpdateStatus": undefined,
    "serial": null,
    "webReputationStatus": "inactive",
  },
  Object {
    "_class": Array [
      "Host",
      "Device",
    ],
    "_key": "trend-micro-computer:35",
    "_type": "trend_micro_computer",
    "agentGUID": undefined,
    "agentStatus": "inactive",
    "antiMalwareStatus": "inactive",
    "applianceStatus": undefined,
    "applicationControlStatus": "inactive",
    "awsAccountId": "697977203433",
    "category": "endpoint",
    "cloudProvider": "AWS",
    "createdOn": undefined,
    "description": "",
    "deviceId": "258AC077-4A7D-F72B-676B-36568A9F8932",
    "displayName": "i-0dca1d41741a0a411",
    "ec2InstanceId": "i-0dca1d41741a0a411",
    "firewallStatus": "inactive",
    "groupId": "trend-micro-computer-group:79",
    "hostGUID": "258AC077-4A7D-F72B-676B-36568A9F8932",
    "hostname": "172.31.211.111",
    "id": "trend-micro-computer:35",
    "integrityMonitoringStatus": "inactive",
    "intrusionPreventionStatus": "inactive",
    "lastSeenOn": undefined,
    "logInspectionStatus": "inactive",
    "make": null,
    "model": null,
    "name": "i-0dca1d41741a0a411",
    "platform": "linux",
    "securityUpdateStatus": undefined,
    "serial": null,
    "webReputationStatus": "inactive",
  },
]
`;
